src/search/index.ts

```diff
@@ -5,8 +5,16 @@ import { Grid, Move } from '../definitions';
 import { alphaBeta } from './alphaBeta';
 import { minimax } from './minimax';
 
+let cache : Map<Grid,Number>;
+
 export const getBestMove = (grid: Grid, forX?: (grid: Grid) => boolean, depth?: number): Move => {
+  
+  
+  if(cache.get(grid) !== undefined){
+    return cache.get(grid);
```

Comment:
Also, we can write this in a single line like, `if (cache.has(grid)) return cache.get(grid);`